### PR TITLE
Add sample extract-only connector

### PR DESCRIPTION
**Pull request by pvanderknyff**
Adds extract-only SQLite connector to show how to make an extract-only connector. Uses ODBC driver.

### CODE REVIEW

File: samples/plugins/sqlite_extract/dialect.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<dialect name='SQLiteODBCExtract'
+         class='sqlite_extract'
+         version='18.1'
+         base='CDataExtractDialect'>
```
Comment (rosswbrown) [inline samples/plugins/sqlite_extract/dialect.xml]: We should consider adding a DefaultExtractDialect and using it in this sample going forward

File: samples/plugins/sqlite_extract/connectionResolver.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version='1.0' encoding='utf-8' ?>
+<tdr class='sqlite_extract'>
+<connection-resolver>
+    <connection-builder>
+        <script file='connectionBuilder.js'/>
+    </connection-builder>
+    <connection-normalizer>
+        <required-attributes>
+            <setImpersonateAttributes/>
+            <attribute-list>
```
Comment (rosswbrown) [inline samples/plugins/sqlite_extract/connectionResolver.xml]: Also need `<attr>authentication</attr>` for the auth none pattern

File: samples/plugins/sqlite_extract/connectionResolver.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+            <attr>server</attr>
+            <attr>one-time-sql</attr>
+            </attribute-list>
+        </required-attributes>
+    </connection-normalizer>
+</connection-resolver>
+
+<driver-resolver>
+    <driver-match >
+      <driver-name type='regex'>SQLite3 ODBC*</driver-name>
```
Comment (rosswbrown) [inline samples/plugins/sqlite_extract/connectionResolver.xml]: Since this is a sample can we do an exact match

**Comment by sidwray** [comment]
Thank you!!

File: samples/plugins/sqlite_extract/manifest.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<?xml version='1.0' encoding='utf-8' ?>
+
+<connector-plugin class='sqlite_extract' superclass='odbc' plugin-version='0.0.0' name='SQLite Extract-Only Taco' version='18.1'>
+   <vendor-information>
+      <company name="Example"/>
```
Comment (rosswbrown) [inline samples/plugins/sqlite_extract/manifest.xml]: From the other samples:
      <company name="Company Name"/>
      <support-link url = "http://example.com"/>

File: samples/plugins/sqlite_extract/manifest.xml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+<?xml version='1.0' encoding='utf-8' ?>
+
+<connector-plugin class='sqlite_extract' superclass='odbc' plugin-version='0.0.0' name='SQLite Extract-Only Taco' version='18.1'>
```
Comment (rosswbrown) [inline samples/plugins/sqlite_extract/manifest.xml]: Remove Taco for samples

File: samples/plugins/sqlite_extract/dialect.xml (hunk 1 above)
Comment (pvanderknyff) [reply samples/plugins/sqlite_extract/dialect.xml]: Sounds like a good idea

**Pull request merged**